collectors/mimecast/utils.js

```diff
@@ -109,6 +128,50 @@ function getAPILogs(authDetails, state, accumulator, maxPagesPerInvocation) {
     });
 }
 
+/**
+   * Unzip the Buffer
+   * 
+   * @param buffer
+   * @returns {Promise}
+   */
+
+function unzipBuffer(buffer) {
+    return new Promise(function (resolve, reject) {
+        var resolved = false;
+        var zip = new AdmZip(buffer);
+        zip.extractAllTo(SIEM_LOGS_PATH, true);
+        // dir path that contains all your json file
+        const files = fs.readdirSync(SIEM_LOGS_PATH);
+        const tempAccumulator = []
+        if (files.length > 0) {
+            files.forEach((file, i) => {
+                if (path.extname(file) === '.json') {
+                    const fileData = JSON.parse(fs.readFileSync(path.join(SIEM_LOGS_PATH, file), 'utf8'));
+                    tempAccumulator.push(...fileData.data);
+                    resolved = true;
+                }
+            });
+            resolve(tempAccumulator);
+        }
+        if (!resolved) {
+            reject(new Error('No files found in zip directory: ' + files.length));
+        }
+    });
+};
+
+function removeTempDirectory(dirPath) {
```

Comment:
I think we need to pass a `callback` parameter into this function in order to make sure we continue execution after folder is cleaned.